mne/channels/interpolation.py

```diff
@@ -251,25 +273,94 @@ def _interpolate_bads_nirs(inst, method="nearest", exclude=(), verbose=None):
     chs = [inst.info["chs"][i] for i in picks_nirs]
     locs3d = np.array([ch["loc"][:3] for ch in chs])
 
-    _check_option("fnirs_method", method, ["nearest", "nan"])
-
-    if method == "nearest":
-        dist = pdist(locs3d)
-        dist = squareform(dist)
-
-        for bad in picks_bad:
-            dists_to_bad = dist[bad]
-            # Ignore distances to self
-            dists_to_bad[dists_to_bad == 0] = np.inf
-            # Ignore distances to other bad channels
-            dists_to_bad[bads_mask] = np.inf
-            # Find closest remaining channels for same frequency
-            closest_idx = np.argmin(dists_to_bad) + (bad % 2)
-            inst._data[bad] = inst._data[closest_idx]
-    else:
-        assert method == "nan"
-        inst._data[picks_bad] = np.nan
+    dist = pdist(locs3d)
+    dist = squareform(dist)
+
+    for bad in picks_bad:
+        dists_to_bad = dist[bad]
+        # Ignore distances to self
+        dists_to_bad[dists_to_bad == 0] = np.inf
+        # Ignore distances to other bad channels
+        dists_to_bad[bads_mask] = np.inf
+        # Find closest remaining channels for same frequency
+        closest_idx = np.argmin(dists_to_bad) + (bad % 2)
+        inst._data[bad] = inst._data[closest_idx]
+
     # TODO: this seems like a bug because it does not respect reset_bads
     inst.info["bads"] = [ch for ch in inst.info["bads"] if ch in exclude]
 
     return inst
+
+
+@verbose
+def _interpolate_bads_seeg(inst, exclude=None, tol=2e-3, verbose=None):
+    if exclude is None:
+        exclude = list()
+    bads_idx = np.zeros(len(inst.ch_names), dtype=bool)
+    goods_idx = np.zeros(len(inst.ch_names), dtype=bool)
+
+    picks = pick_types(inst.info, meg=False, seeg=True, exclude=exclude)
+    inst.info._check_consistency()
+    bads_idx[picks] = [inst.ch_names[ch] in inst.info["bads"] for ch in picks]
+
+    if len(picks) < 3 or bads_idx.sum() == 0:
+        return
+
+    goods_idx[picks] = True
+    goods_idx[bads_idx] = False
+
+    pos = inst._get_channel_positions(picks)
+
+    # Make sure only sEEG are used
+    bads_idx_pos = bads_idx[picks]
+
+    # for each bad contact:
+    # 1) find nearest neighbor to define the electrode shaft line
+    # 2) find all contacts on the same line
+    # 3) interpolate the bad contacts
+
+    dist = squareform(pdist(pos))
+    np.fill_diagonal(dist, np.inf)
+
+    picks_bad = list(np.where(bads_idx_pos)[0])
+    while picks_bad:
+        bad = picks_bad[0]
+        n1 = pos[bad]
+        n2 = pos[np.argmin(dist[bad])]  # 1
+        # https://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html
+        shaft_dists = np.array(
+            [
+                0
+                if all(n0 == n1) or all(n0 == n2)
+                else np.linalg.norm(np.cross((n0 - n1), (n0 - n2)))
+                / np.linalg.norm(n2 - n1)
+                for n0 in pos
+            ]
+        )
```

Comment:
here again, my initial reaction is that better var names / intermediate vars with descriptive names could help make this more maintainable.  But looking closer I think that will only work if you convert to a proper for-loop rather than a list comprehension?  I guess a helper func that was called via `apply_along_axis` could also work, but seems like overkill; a plain for-loop is probably not too slow here since we're dealing with tens or at most a few hundreds of contacts (right)?

For example, IIUC `all(n0 == n1)` means "this electrode is the current bad electrode (the one picked out by `bad = picks_bad[0]`)" which would have been much easier to grok if I saw something like

```py
for _pos in pos:
    is_current_bad_elec = all(_pos == pos[bad])
    is_closest_bad_elec = all(_pos == pos[np.argmin(dist[bad])])
    ...
```